### ISSUE TITLE
Error

### ISSUE THREAD

**Issue by DJohnston79** [chmurph2/Detect-Displays#1]
error "System Events got an error: Can’t get process \"System Preferences\"." number -1728 from process "System Preferences"

**Comment by chmurph2** [comment]
Hey @DJohnston79 thanks for opening this issue.

My guess as to what is wrong is "System Preferences" has been renamed to "System Settings" in recent versions of macOS.
After that, I suspect other steps will be broken, due to changes [like this](https://support.apple.com/guide/mac-help/mac-recognize-external-display-mchla25b377d/mac):

> Press and hold the Option key to make the Detect Displays button appear in place of the Night Shift button, then click Detect Displays.

I'm no longer using this Applescript but am happy to accept and test any pull requests.

**Comment by chmurph2** [comment]
I did give it a couple minutes and looks like there will also be assistive permissions issues to figure out.

![CleanShot 2024-04-26 at 09 14 27@2x](https://github.com/chmurph2/Detect-Displays/assets/2841/e97ad089-7a69-49ca-8712-c4445b210314)


I'm not sure, but perhaps approaching this with [a Shortcut](https://support.apple.com/guide/shortcuts-mac/welcome/mac) would be better.

**Comment by DJohnston79** [comment]
Yeah, I worked out the name change and it seemed to move a bit more .  I
think I got the logic worked out now . Thanks for replying!

**Comment by WangSoooCute** [comment]
@DJohnston79 Hey! I have came across the same problem today. How did you solve that?

**Comment by DJohnston79** [comment]
> @DJohnston79 Hey! I have came across the same problem today. How did you solve that?

Rename system preferences to system settings or settings I believe :)

**Comment by chmurph2** [comment]
If anyone has a working version I'd be happy to test out a pull request :).

**Comment by DJohnston79** [comment]
I got this so far:

`ell application "System Settings" to activate

delay 0.5

tell application "System Events"
	tell process "System Settings"
		click menu item "Displays" of menu "View" of menu bar 1
		try --don't even consider not using a try block!
			key down option
			delay 1.0
			click button "Detect Displays" of window 1
			delay 0.2
			key up option
		on error --logging out is the only other way to clear these
			key up option
		end try
	end tell
end tell`

**Comment by DJohnston79** [comment]
I have this :

[https://pastebin.com/uJirSG2W](url)
